src/tests/test_graph_base.cc

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright 2009-2018 The VOTCA Development Team (http://www.votca.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#define BOOST_TEST_MAIN
+
+#define BOOST_TEST_MODULE graph_test
+#include <boost/test/unit_test.hpp>
+#include <cmath>
+#include <exception>
+#include <iostream>
+#include <iostream>
+#include <votca/tools/edge.h>
+#include <votca/tools/graph.h>
+#include <votca/tools/graphnode.h>
+using namespace std;
+using namespace votca::tools;
+
+// used for rounding doubles so we can compare them
+double round_(double v, int p) {
+  v *= pow(10, p);
+  v = round(v);
+  v /= pow(10, p);
+  return v;
+}
+
+BOOST_AUTO_TEST_SUITE(graph_test)
+
+BOOST_AUTO_TEST_CASE(constructors_test) { Graph g; }
+
+BOOST_AUTO_TEST_CASE(isolatednodes_test) {
```

Comment:
please add some doc with information what do you plan to test in the unit

Reply:
Good point.